devolving_music/lib/song_scores.py

```diff
@@ -29,18 +31,35 @@ def get_scores(self) -> "dict[int, ScoreSuite]":
 
         return self.song_score_dict
 
+    def get_scores_list(self) -> "list[ScoreSuite]":
+        # calculates scores for all submissions using current comparisons
+        # returns list of all score suite objects
+        return list(self.get_scores().values())
+
     def get_compare_submission_random(self, submission_id) -> SongSubmission:
         key_list = list(self.song_score_dict.keys())
         key_list.remove(submission_id)
-        # once you have a critical number of comparisons then pull from quality
-        # list
-        return self.song_score_dict.get(random.choice(key_list)).song_submission
+        return self.song_score_dict.get(
+            random.choice(key_list)).song_submission
+
+    def get_compare_submission_closest(self, score_suite_obj: "ScoreSuite", information_threshold : int, update_scores = False):
+        # information threshold is the threshhold at which we start to consider score_suites as being able to be close to score_suite_obj
+        # score_suites below this information threshhold are considered to be infinitely far away from score_suite_obj
+        if(update_scores):
+            self.get_scores()
```

Comment:
Why put that into this method? It seems unrelated, make it the job of the outside context to update scores if it wants to.